skimage/feature/fisher_vector.py

```diff
@@ -0,0 +1,223 @@
+
+    References
+    -----------
+    .. [1] https://scikit-learn.org/stable/modules/generated/sklearn.mixture.GaussianMixture.html
+
+    Examples
+    --------
+    >>> from skimage.feature import fisher_vector
+    >>> sift_for_images = [np.random.random((10, 128)) for _ in range(10)]
+    >>> num_modes = 16
+    >>> # Estimate 16-mode GMM with these synthetic SIFT vectors
+    >>> gmm = fisher_vector.learn_gmm(sift_for_images, num_modes)
+    """
+
+    try:
+        from sklearn.mixture import GaussianMixture
+    except ImportError:
+        raise ImportError(
+            'scikit-learn is not installed. Please ensure it is installed in '
+            'order to use the Fisher vector functionality.'
+        )
+
+    if not isinstance(descriptors, (list, np.ndarray)):
+        raise DescriptorException(
+            'Please ensure descriptors are either a NumPY array, '
+            'or a list of NumPy arrays.'
+        )
+
+    d_mat_1 = descriptors[0]
+    if isinstance(descriptors, list) and not isinstance(d_mat_1, np.ndarray):
+        raise DescriptorException(
+            'Please ensure descriptors are a list of NumPy arrays.'
+        )
+
+    if isinstance(descriptors, list):
+        expected_shape = descriptors[0].shape
+        ranks = [len(e.shape) == len(expected_shape) for e in descriptors]
+        if (not all(ranks)):
+            raise DescriptorException(
+                'Please ensure all elements of your descriptor list '
+                'are of rank 2.'
+            )
+        dims = [e.shape[1] == descriptors[0].shape[1] for e in descriptors]
+        if not all(dims):
+            raise DescriptorException(
+                'Please ensure all descriptors are of the same dimensionality.'
+            )
+
+    if not isinstance(num_modes, int) or num_modes <= 0:
+        raise FisherVectorException(
+            'Please ensure num_modes is a positive integer.'
+        )
+
+    if 'covariance_type' in kwargs and kwargs['covariance_type'] != 'diag':
+        raise FisherVectorException('Covariane type must be "diag".')
+
+    if isinstance(descriptors, list):
+        descriptors = np.vstack(descriptors)
+
+    gmm = GaussianMixture(n_components=num_modes, covariance_type='diag')
+
+    gmm.fit(descriptors)
+
+    return gmm
+
+
+def compute_fisher_vector(descriptors, gmm, improved=False):
+    """Function to compute the Fisher vector given some descriptors/vectors,
+    and an associated estimated GMM.
+
+    Parameters
+    -----------
+    descriptors : np.ndarray
+        NumPy array of the descriptors to use to compute the Fisher vector
+        representation for.
+    gmm : sklearn.mixture.GaussianMixture
+        An estimated GMM object, which contains the necessary parameters needed
+        to compute the Fisher vector.
+    improved : bool, default=False
+        Flag denoting when to compute improved Fisher vectors or not. Improved
+        Fisher vectors are L2 and power normalized.
```

Comment:
What does power normalized mean?  You can either expand this explanation, or add a Notes section to the docstring.

Reply:
I'll expand in here

Reply:
Done